### PR TITLE
Add TryGetSystemEventData

### LINKED ISSUE #44663: Add extension method TryGetSystemEvent to Azure.Messaging.EventGrid.SystemEvents prior to GA
See https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventgrid/Azure.Messaging.EventGrid/src/EventGridExtensions.cs#L25

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/44663

### CODE REVIEW

File: sdk/eventgrid/Azure.Messaging.EventGrid.SystemEvents/src/CloudEventExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+using System;
+using System.Text.Json;
+
+namespace Azure.Messaging.EventGrid.SystemEvents
+{
+    /// <summary>
+    /// Extension methods for BinaryData to use for parsing JSON-encoded events.
+    /// </summary>
+    public static class CloudEventExtensions
```
Comment (JoshLove-msft) [inline sdk/eventgrid/Azure.Messaging.EventGrid.SystemEvents/src/CloudEventExtensions.cs:12]: I decided to rename this type from the [EventGrid version](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventgrid/Azure.Messaging.EventGrid/src/EventGridExtensions.cs) and forego the type forwarding. This way we could leave everything in the SystemEvents namespace.

File: sdk/eventgrid/Azure.Messaging.EventGrid.SystemEvents/README.md [hunk 2]
```diff
@@ -21,38 +21,89 @@ dotnet add package Azure.Messaging.EventGrid.SystemEvents --prerelease
...
-```
-
-#### Authenticate using Topic Access Key
-
-Once you have your access key and topic endpoint, you can create the publisher client as follows:
-```C# Snippet:CreateNamespaceClient
-// Construct the client using an Endpoint for a namespace as well as the shared access key
-var senderClient = new EventGridSenderClient(new Uri(namespaceTopicHost), topicName, new AzureKeyCredential(namespaceKey));
+#### Deserializing event data
+From here, one can access the event data by deserializing to a specific type by calling `ToObjectFromJson<T>()` on the `Data` property. In order to deserialize to the correct type, the `EventType` property (`Type` for CloudEvents) helps distinguish between different events. Custom event data should be deserialized using the generic method `ToObjectFromJson<T>()`. There is also an extension method `ToObject<T>()` that accepts a custom `ObjectSerializer` to deserialize the event data.
```
Comment (jsquire) [inline sdk/eventgrid/Azure.Messaging.EventGrid.SystemEvents/README.md:37]:
```suggestion

From here, one can access the event data by deserializing to a specific type by calling `ToObjectFromJson<T>()` on the `Data` property. In order to deserialize to the correct type, the `EventType` property (`Type` for CloudEvents) helps distinguish between different events. Custom event data should be deserialized using the generic method `ToObjectFromJson<T>()`. There is also an extension method `ToObject<T>()` that accepts a custom `ObjectSerializer` to deserialize the event data.
```

File: sdk/eventgrid/Azure.Messaging.EventGrid.SystemEvents/README.md [hunk 3]
```diff
@@ -21,38 +21,89 @@ dotnet add package Azure.Messaging.EventGrid.SystemEvents --prerelease
 
 You must have an [Azure subscription](https://azure.microsoft.com/free/dotnet/) and an Azure resource group with a custom Event Grid topic or domain. Follow this [step-by-step tutorial](https://docs.microsoft.com/azure/event-grid/custom-event-quickstart-portal) to register the Event Grid resource provider and create Event Grid topics using the [Azure portal](https://portal.azure.com/). There is a [similar tutorial](https://docs.microsoft.com/azure/event-grid/custom-event-quickstart) using [Azure CLI](https://docs.microsoft.com/cli/azure).
 
-### Authenticate the client
+### Receiving and Deserializing Events
+There are several different Azure services that act as [event handlers](https://docs.microsoft.com/azure/event-grid/event-handlers).
```
Comment (jsquire) [inline sdk/eventgrid/Azure.Messaging.EventGrid.SystemEvents/README.md:25]:
```suggestion

There are several different Azure services that act as [event handlers](https://docs.microsoft.com/azure/event-grid/event-handlers).
```

File: sdk/eventgrid/Azure.Messaging.EventGrid.SystemEvents/README.md [hunk 4]
```diff
@@ -21,38 +21,89 @@ dotnet add package Azure.Messaging.EventGrid.SystemEvents --prerelease
...
-#### Authenticate using an Entra ID
-
-Azure Event Grid provides integration with Entra ID for identity-based authentication of requests. With Entra ID, you can use role-based access control (RBAC) to grant access to your Azure Event Grid resources to users, groups, or applications. The [Azure Identity library](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/identity/Azure.Identity/README.md) provides easy Azure Active Directory support for authentication.
-
-To send events to a topic or domain using Azure Active Directory, the authenticated identity should have the "EventGrid Data Sender" role assigned.
-
-```C# Snippet:CreateNamespaceClientAAD
-// Construct the sender client using an Endpoint for a namespace as well as the DefaultAzureCredential
-var senderClient = new EventGridSenderClient(new Uri(namespaceTopicHost), topicName, new DefaultAzureCredential());
+Using `TryGetSystemEventData()`:
```
Comment (jsquire) [inline sdk/eventgrid/Azure.Messaging.EventGrid.SystemEvents/README.md:63]: Should this be a header of some level?

**Review by jsquire — APPROVED** [review]

File: sdk/eventgrid/Azure.Messaging.EventGrid.SystemEvents/README.md (hunk 2 above)
Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid.SystemEvents/README.md:37]: FYI I don't think this impacts the rendered markdown

File: sdk/eventgrid/Azure.Messaging.EventGrid.SystemEvents/src/CloudEventExtensions.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid.SystemEvents/src/CloudEventExtensions.cs:12]: Thinking about this more, I don't think we want to do this as we will end up with duplicate copies of this extension method.

**Pull request merged**